### PR TITLE
Non-human (Nuke\Clown) Ops

### PR DESCRIPTION
**Pull request by RaveRadbury**
## About The Pull Request
Back when plasmamen were added, #16075 was merged to prevent them from spawning as nukies to keep them from bursting into flames when they spawned. Before this lizard ops were possible (we didn't have any other non-humans at the time).

This PR restores non-human ops and makes a compromise for plasmamen.

## Why It's Good For The Game
* It doesn't make sense from a lore perspective for Op teams to be human only. 
* Players will enjoy the challenge of playing Op as a non-human
* Some players just like playing as certain species

## Changelog
:cl:
balance: Nuke-Ops will now spawn as the species that they rolled with.
/:cl:

Permission for this granted by @Mothblocks
Approved as Lore-Friendly by @EOBGames

### CODE REVIEW

**Comment by TheSilverNuke** [comment]
I thought only Nanotrasen was super inclusive with species. Also what do you do with Plasmamen?

**Comment by RaveRadbury** [comment]
> I thought only Nanotrasen was super inclusive with species. Also what do you do with Plasmamen?

I've seen stuff claiming that the syndicate is more inclusive, I ran this past EOB first (and I'll add that to the PR body)

Plasmamen spawn in their nukie suit, with a tank of plasma.

**Comment by TemporalOroboros** [comment]
> 
> 
> I thought only Nanotrasen was super inclusive with species. Also what do you do with Plasmamen?

Other way around. Nanotrasen is supposed to be really speciesist and the syndicate is supposed to be the inclusive one. At least that's what I heard the last time I checked the lore.

**Comment by Farquaar** [comment]
> 
> 
> > I thought only Nanotrasen was super inclusive with species. Also what do you do with Plasmamen?
> 
> Other way around. Nanotrasen is supposed to be really speciesist and the syndicate is supposed to be the inclusive one. At least that's what I heard the last time I checked the lore.

That's new, then. Classically speaking, Nanotrasen pushed for the acceptance for other species so they could benefit from the cheap labour.

**Comment by Mothblocks** [comment]
the lore master already ok'd this pr

**Comment by RaveRadbury** [comment]
@Farquaar I think the line drawn was specifically `the syndicate will entrust non-humans with tasks and duties that NT won't` especially re: NT's enforced human authority.

**Comment by zxaber** [comment]
The syndicate is a collection of companies and interest groups that share NT as a common enemy. It seems probable that at least one of those groups is made of (or at least tolerates) non-human humanoids, and I wouldn't imagine the syndicate is willing to turn down volunteers for a suicidal mission even if the leadership is biased against a particular species.

**Comment by tralezab** [comment]
sure why not

**Comment by anconfuzedrock** [comment]
For those who didn't know, years ago nuke ops could be nonhumans and the only reason it got removed was because when plasmamen were added they caught fire spawning as nuke op. It was just easier to remove nonhuman nuke ops than to fix the plasmaman issue: #16075

File: code/modules/antagonists/nukeop/nukeop.dm [hunk 1]
```diff
@@ -26,8 +26,9 @@
 	if(!ishuman(owner.current))
 		return
 	var/mob/living/carbon/human/H = owner.current
-
-	H.set_species(/datum/species/human) //Plasamen burn up otherwise, and lizards are vulnerable to asimov AIs
+	
+	if(isplasmaman(H))
+		nukeop_outfit = text2path("[nukeop_outfit]/plasmaman")
```
Comment (Mothblocks) [inline code/modules/antagonists/nukeop/nukeop.dm:31]: NOOOOOO

Comment (Mothblocks) [reply code/modules/antagonists/nukeop/nukeop.dm:31]: Have a plasmaman variable instead. Do NOT use text paths, ever.

File: code/modules/antagonists/nukeop/outfits.dm [hunk 2]
```diff
@@ -63,3 +63,28 @@
 		/obj/item/tank/jetpack/oxygen/harness=1,\
 		/obj/item/gun/ballistic/automatic/pistol=1,\
 		/obj/item/kitchen/knife/combat/survival)
+
+/datum/outfit/syndicate/plasmaman
+	name = "Syndicate Operative - Plasmaman"
+	mask = /obj/item/clothing/mask/gas/syndicate
+	suit = /obj/item/clothing/suit/space/hardsuit/syndi
+	r_hand= /obj/item/tank/internals/plasmaman/belt/full
```
Comment (Mothblocks) [inline code/modules/antagonists/nukeop/outfits.dm:71]:
```suggestion
	r_hand = /obj/item/tank/internals/plasmaman/belt/full
```

File: code/modules/antagonists/nukeop/outfits.dm [hunk 3]
```diff
@@ -63,3 +63,28 @@
...
+	backpack_contents = list(/obj/item/kitchen/knife/combat/survival)
+
+/datum/outfit/syndicate/leader/plasmaman
+	name = "Syndicate Leader - Plasmaman"
+	mask = /obj/item/clothing/mask/gas/syndicate
+	suit = /obj/item/clothing/suit/space/hardsuit/syndi
+	r_hand= /obj/item/tank/internals/plasmaman/belt/full
```
Comment (Mothblocks) [inline code/modules/antagonists/nukeop/outfits.dm:78]:
```suggestion
	r_hand = /obj/item/tank/internals/plasmaman/belt/full
```

File: code/modules/antagonists/nukeop/outfits.dm [hunk 4]
```diff
@@ -63,3 +63,28 @@
...
+	gloves = /obj/item/clothing/gloves/krav_maga/combatglovesplus
+	command_radio = TRUE
+
+	id_trim = /datum/id_trim/chameleon/operative/nuke_leader
+
+/datum/outfit/syndicate/no_crystals/plasmaman
+	name = "Syndicate Operative - Plasmaman Reinforcement"
+	mask = /obj/item/clothing/mask/gas/syndicate
+	suit = /obj/item/clothing/suit/space/hardsuit/syndi
+	r_hand= /obj/item/tank/internals/plasmaman/belt/full
```
Comment (Mothblocks) [inline code/modules/antagonists/nukeop/outfits.dm:89]:
```suggestion
	r_hand = /obj/item/tank/internals/plasmaman/belt/full
```

File: code/modules/antagonists/clown_ops/outfits.dm [hunk 5]
```diff
@@ -29,3 +29,28 @@
...
+		/obj/item/dnainjector/clumsymut, //in case you want to be clumsy for the memes
+		/obj/item/storage/box/syndie_kit/clownpins, //for any guns that you get your grubby little clown op mitts on
+		/obj/item/reagent_containers/spray/waterflower/lube)
+
+
+/datum/outfit/syndicate/clownop/leader/plasmaman
+	name = "Clown Operative Leader - Plasmaman"
+	uniform = /obj/item/clothing/under/plasmaman/clown
+	head = /obj/item/clothing/head/helmet/space/plasmaman/clown
+	r_hand= /obj/item/tank/internals/plasmaman/belt/full
```
Comment (Mothblocks) [inline code/modules/antagonists/clown_ops/outfits.dm:48]:
```suggestion
	r_hand = /obj/item/tank/internals/plasmaman/belt/full
```

File: code/modules/antagonists/clown_ops/outfits.dm [hunk 6]
```diff
@@ -29,3 +29,28 @@
 	command_radio = TRUE
 
 	id_trim = /datum/id_trim/chameleon/operative/clown_leader
+
+/datum/outfit/syndicate/clownop/plasmaman
+	name = "Clown Operative - Plasmaman"
+	uniform = /obj/item/clothing/under/plasmaman/clown
+	head = /obj/item/clothing/head/helmet/space/plasmaman/clown
+	r_hand= /obj/item/tank/internals/plasmaman/belt/full
```
Comment (Mothblocks) [inline code/modules/antagonists/clown_ops/outfits.dm:37]:
```suggestion
	r_hand = /obj/item/tank/internals/plasmaman/belt/full
```

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Comment by Cyprex** [comment]
I think the balance implications of this are considerable, especially for plasmamen

**Comment by SkeletalElite** [comment]
I mean I play a human character anyways, but I feel like if I was playing as a plasma man I'd rather be swapped to human when rolling nuke ops, but maybe that's just me.

File: code/modules/antagonists/nukeop/nukeop.dm (hunk 1 above)
Comment (tralezab) [reply code/modules/antagonists/nukeop/nukeop.dm:31]: hahahaha oh noooooooo

**Comment by Dorsisdwarf** [comment]
Isn't this the sort of thing that Backup Human Character is a pref for?

**Comment by RaveRadbury** [comment]
> Isn't this the sort of thing that Backup Human Character is a pref for?

You'd think so, but currently they just take the random of the species you rolled and then set it to human.

**Comment by RaveRadbury** [comment]
> #16075

stealing this btw

File: code/modules/antagonists/nukeop/nukeop.dm [hunk 7]
```diff
@@ -26,8 +26,9 @@
 	if(!ishuman(owner.current))
 		return
 	var/mob/living/carbon/human/H = owner.current
-
-	H.set_species(/datum/species/human) //Plasamen burn up otherwise, and lizards are vulnerable to asimov AIs
+	
+	if(isplasmaman(H))
```
Comment (coiax) [inline code/modules/antagonists/nukeop/nukeop.dm:30]: Mmm, I don't really like how this uses a check in the antagonist equipping step.

Is there some way we can use some hook in the `/datum/species` datums, or create one?

Job code already has a [hook ](https://github.com/tgstation/tgstation/blob/master/code/modules/mob/living/carbon/human/species_types/plasmamen.dm#L81)in /datum/species.

**Comment by Krysonism** [comment]
I liked that there is a bit of contrast between NT and the syndicate. 

Is it really that hard to not  play your catgirl static once in a blue moon?

**Comment by Bawhoppen** [comment]
Maybe make it so the lieutenants can't be nonhuman? I still have the feeling that much of the Syndicate would be a bit racist, even if they're willing to send a few felinids on suicide missions. Probably not keen on commissioned officers being non-human.

**Comment by ilikesaying** [comment]
can this be a toggle? I like being a human nukeops due to the lack of disadvantages it has (no damage multipliers, no vulnerabilities to things)

**Comment by Mothblocks** [comment]
Maybe after new prefs menu, but definitely not now.

File: code/modules/antagonists/clown_ops/outfits.dm [hunk 8]
```diff
@@ -29,3 +29,28 @@
...
 
 	id_trim = /datum/id_trim/chameleon/operative/clown_leader
+
+/datum/outfit/syndicate/clownop/plasmaman
+	name = "Clown Operative - Plasmaman"
+	uniform = /obj/item/clothing/under/plasmaman/clown
+	head = /obj/item/clothing/head/helmet/space/plasmaman/clown
+	r_hand = /obj/item/tank/internals/plasmaman/belt/full
+	backpack_contents = list(/obj/item/kitchen/knife/combat/survival,
+		/obj/item/dnainjector/clumsymut, //in case you want to be clumsy for the memes
```
Comment (ATH1909) [inline code/modules/antagonists/clown_ops/outfits.dm:39]: clown op plasmaman can't give themselves the clumsiness mutation with their dna mutator, so maybe you should just remove it from their outfit

File: code/modules/antagonists/nukeop/nukeop.dm (hunk 7 above)
Comment (ATH1909) [reply code/modules/antagonists/nukeop/nukeop.dm:30]: ain't there some variable or something for dangerous species?

**Comment by ATH1909** [comment]
> I mean I play a human character anyways, but I feel like if I was playing as a non numan I'd rather be swapped to human when rolling nuke ops, but maybe that's just me.

maybe we could add a preference for this?

File: code/modules/antagonists/clown_ops/outfits.dm (hunk 8 above)
Comment (RaveRadbury) [reply code/modules/antagonists/clown_ops/outfits.dm:39]: Thanks for the catch, @ATH1909

**Comment by coiax** [comment]
We should not make everything a preference.

**Comment by RaveRadbury** [comment]
> We should not make everything a preference.

There have been conversations in the past about a preference to make antags a random human on spawn. It might be something to consider for the future, but at the same time we should iron out this wrinkle, the only roundstart antag that forces you into playing human.

**Comment by RaveRadbury** [comment]
Gonna give the code some tlc tomorrow I think.

**Comment by Technoturnovers** [comment]
I don't object to non-human nuke ops, but I feel like there should be some setting to force human when you're antag, that way people who don't want to accidentally handicap themselves re: plasmaman can prevent that from happening

**Comment by RaveRadbury** [comment]
> I don't object to non-human nuke ops, but I feel like there should be some setting to force human when you're antag, that way people who don't want to accidentally handicap themselves re: plasmaman can prevent that from happening

This is something that has been discussed on the forums in the past, specifically using it in conjunction with the "randomize on antag" feature.

**Comment by CMDR-Gungnir** [comment]
> > I don't object to non-human nuke ops, but I feel like there should be some setting to force human when you're antag, that way people who don't want to accidentally handicap themselves re: plasmaman can prevent that from happening
> 
> This is something that has been discussed on the forums in the past, specifically using it in conjunction with the "randomize on antag" feature.

Perhaps this could be fixed with a Magic Mirror type thing in the Nukie Base? Since Nukie Names are just Codenames anyway, who cares that Kyler Fancypants is a Plasmaman, he's just concealing his real name.

**Comment by RaveRadbury** [comment]
I swear I'm working on this still I just had a big IRL thing.

**Comment by RaveRadbury** [comment]
I swear I'm working on this still I just had a lot of IRL to deal with recently.

**Comment by RaveRadbury** [comment]
With many appreciative thanks to the maintainer team for keeping an eye on this while I had to go and get smokes from the corner store I will now get on this, right... now.

**Comment by RaveRadbury** [comment]
Clownops were conflicting for some reason but it wouldn't show why 🤷

**Comment by LemonInTheDark** [comment]
@coiax @RaveRadbury what's the jive?

**Comment by RaveRadbury** [comment]
> @coiax @RaveRadbury what's the jive?

I'm still trying to figure out what was meant in the feedback re: hooks and also how I'm supposed to go about doing this without the species check (mind an ishuman check happens a few lines above it)

@coiax and I have been at odds in terms of time zones and schedules

I'd like to think I could get this knocked out in a day but I'm hitting a wall on this stuff.

**Comment by RaveRadbury** [comment]
gonna close this for a minute

**Pull request closed without merging**